addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -129,8 +129,7 @@ export class MassMailingHtmlField extends HtmlField {
             // Wait for the css and images to be loaded.
             await iframePromise;
             const editableClone = iframe.contentDocument.querySelector('.note-editable');
-            this.cssRules = this.cssRules || getCSSRules($editable[0].ownerDocument);
-            await toInline($(editableClone), this.cssRules, $(iframe));
+            await toInline($(editableClone), this.wysiwyg._rulesCache, $(iframe));
```

Comment:
I believe you're correct that `this.cssRules` shouldn't have been defined here. I notice that in 15.0, this instance variable is _passed_ to `toInline` in two places, _but never defined_. So it's effectively useless. Then we [defined it](https://github.com/odoo/odoo/commit/1173733ba7c33abf079d668ea219e316401352d3#diff-0b4e81825324dc36e89543232f3f8f8704acddc4ec21031c6ccea78ee93b00eeR127), and we end up with two distinct caches: `HtmlField.cssRules` and `Wysiwyg._rulesCache`.
Now, `toInline` already does this:
```js
cssRules = cssRules || wysiwyg && wysiwyg._rulesCache;
if (!cssRules) {
    cssRules = getCSSRules(doc);
    if (wysiwyg) {
        wysiwyg._rulesCache = cssRules;
    }
}
```
so _as long as `toInline` can find `wysiwyg`_, passing it `this.wysiwyg._rulesCache` should not have any effect.

In conclusion, I would remove any reference to `this.cssRules` anywhere in the codebase and replace it with `undefined`. Then check if everything works. If it doesn't it's likely because `toInline` can't find `wysiwyg` at some point. But I suspect it will work :-)

```suggestion
            await toInline($(editableClone), undefined, $(iframe));
```

cc @dmo-odoo

Reply:
@Zynton Thanks for the suggestion, I updated the change , can you take a look when you have time ? thanks

Reply:
LGTM. @dmo-odoo ?

Reply:
Did one of you test that the cache was properly used when it should? That is we should not recompute the rules twice when calling `toInline` twice if we didn't change anything in the design tab.

Reply:
I didn't think about this , I will test it and get back to you

Reply:
@dmo-odoo , I changed the fix here for stable, and I will update it in master to pass options into `toInline`